onnxscript/rewriter/pattern.py

```diff
@@ -1,5 +1,58 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
+"""Pattern-based rewriter API for ONNX models.
+
+This module provides the main user-facing API for the ONNX pattern rewriter.
+It allows users to define patterns that match subgraphs in ONNX models and
+replace them with more efficient implementations.
+
+Example Usage:
```

Comment:
Can you illustrate usage with class-based rules? It is better to encourage users to use them instead of the lower level primitive illustrated here.